hyco-https/lib/HybridConnectionHttpsServer.js

```diff
@@ -376,20 +375,42 @@ function connectControlChannel(server) {
   // This represents the token renew timer/interval, keep a reference in order to cancel it.
   var tokenRenewTimer = null;
 
+  // KeepAlive interval to detect half-open connections
+  var keepAliveTimer = null;
+
   server.controlChannel.onerror = function(event) {
     server.emit('error', event);
-    clearInterval(tokenRenewTimer);
+    clearIntervals();
     if (!server.closeRequested) {
       connectControlChannel(server);
     }
   }
 
   server.controlChannel.onopen = function(event) {
     server.emit('listening');
+
+    var keepAliveInterval = null;
+    try {
+      keepAliveInterval = server.options.keepAliveTimeout.asMilliseconds();
+    } catch (ex) {
+      console.log("keepAliveTimeout should be an instance of moment.duration");
+    }
+    if (keepAliveInterval) {
+      keepAliveTimer = setInterval(function() {
+        try {
+          server.controlChannel.pong();
+        } catch (e) {
+          if (!server.closeRequested) {
+            // Closed unexpectedly, should reconnect
+            connectControlChannel(server);
+          }
```

Comment:
Should clearIntervals be called here?

Reply:
Yes that's correct

---
In reply to: [315839091](https://github.com/Azure/azure-relay-node/pull/34#discussion_r315839091) [](ancestors = 315839091)